operator/pkg/name/name_test.go

```diff
@@ -125,3 +126,108 @@ a:
 		})
 	}
 }
+
+func TestRootNamespaceForMeshConfig(t *testing.T) {
+	tests := []struct {
+		desc              string
+		yamlStr           string
+		wantRootNamespace string
+	}{
+		{
+			desc: "only iopSpec.Namespace",
+			yamlStr: `
+namespace: istio-foo`,
+			wantRootNamespace: "istio-foo",
+		},
+		{
+			desc: "only iopSpec.Values[global].istioNamespace",
+			yamlStr: `
+values:
+  global:
+    istioNamespace: istio-foo`,
+			wantRootNamespace: "istio-foo",
+		},
+		{
+			desc: "only iopSpec.Values[meshConfig].rootNamespace",
+			yamlStr: `
+values:
+  meshConfig:
+    rootNamespace: istio-foo`,
+			wantRootNamespace: "istio-foo",
+		},
+		{
+			desc: "only iopSpec.MeshConfig.rootNamespace",
+			yamlStr: `
+meshConfig:
+  rootNamespace: istio-foo`,
+			wantRootNamespace: "istio-foo",
+		},
+		{
+			desc: "iopSpec.Values[meshConfig].rootNamespace and iopSpec.Values[meshConfig].rootNamespace",
+			yamlStr: `
+values:
+  global:
+    istioNamespace: istio-foo
+  meshConfig:
+    rootNamespace: istio-bar`,
+			wantRootNamespace: "istio-bar",
+		},
+		{
+			desc: "iopSpec.Values[meshConfig].rootNamespace and iopSpec.MeshConfig.rootNamespace",
+			yamlStr: `
+meshConfig:
+  rootNamespace: istio-bar
+values:
+  global:
+    istioNamespace: istio-foo`,
+			wantRootNamespace: "istio-bar",
+		},
+		{
+			desc: "iopSpec.Namespace, iopSpec.Values[meshConfig].rootNamespace and iopSpec.Values[meshConfig].rootNamespace",
+			yamlStr: `
+namespace: istio-foo
+values:
+  global:
+    istioNamespace: istio-foo
+  meshConfig:
+    rootNamespace: istio-bar`,
+			wantRootNamespace: "istio-bar",
+		},
+		{
+			desc: "iopSpec.Namespace, iopSpec.Values[meshConfig].rootNamespace and iopSpec.MeshConfig.rootNamespace",
+			yamlStr: `
+namespace: istio-foo
+meshConfig:
+  rootNamespace: istio-bar
+values:
+  global:
+    istioNamespace: istio-foo`,
+			wantRootNamespace: "istio-bar",
+		},
+		{
+			desc: "iopSpec.MeshConfig.rootNamespace and iopSpec.Values[meshConfig].rootNamespace",
+			yamlStr: `
+meshConfig:
+  rootNamespace: istio-bar
+values:
+  global:
+    istioNamespace: istio-foo
+  meshConfig:
+    rootNamespace: istio-bar`,
+			wantRootNamespace: "istio-bar",
```

Comment:
could you add a test that shows that `spec.meshConfig.rootNamespace` takes precedence over `spec.values.meshConfig.rootNamespace`?

Reply:
thank, added case.